### ISSUE TITLE
test: Print file path of a test file that fails outside any running tests.

### ISSUE THREAD

**Issue by jespertheend** [denoland/deno#14312]
For example:
```js
const e = new Error("oh no!");
e.stack = "stack is gone";
throw e;

Deno.test({
    name: "fails outside the test",
    fn() {},
});
```

prints the following:

![image](https://user-images.githubusercontent.com/2737650/163821044-0c2abff7-fbd1-4ff2-8080-02d6cd34efc5.png)

Which makes it hard to figure out the error originated from `c.test.js`.

Admittedly I cheated a little here by removing the stack trace of the error, which would have shown `c.test.js` in the stack trace, so a more real world use case would be:

```js
import {Importer} from "https://deno.land/x/fake_imports@v0.4.0/mod.js";

const importer = new Importer(import.meta.url);
await importer.import("does/not/exist.js");

Deno.test({
    name: "fails outside the test",
    fn() {},
});
```

![image](https://user-images.githubusercontent.com/2737650/163821399-31b3205b-4b11-48fd-bffb-31ed128043e7.png)

The rejected promise is created in a file other than the running test, causing the stack trace to never mention `c.test.js`.

At the moment the only way to figure out where the error originated is to gradually reduce the scope of the tests you're running, which could take pretty long depending on how many tests you have.

**Comment by bartlomieju** [comment]
Thanks for opening this issue. I discussed it last week with @dsherret that it's very unhelpful. I have an idea how to fix this, but that requires making test runner "more rusty", by moving some of the logic from JavaScript to Rust. It will be a bigger refactor, but we should tackle it.

**Comment by nayeemrmn** [comment]
Fixed in #14513.
